search-api.md

```diff
@@ -230,3 +224,27 @@ A synchronous `application/json` response, of the following form:
 #### Results
 * ***Mandatory***, but can be empty
 * Is a **list of matches**, where each match has a `patient` object of the same format as the one described above for the query
+
+
+### Error handling
+The remote server should use HTTP status codes to report any errors encoundered processing the match request. Here are a list of status codes and their meanings with regards to this API:
+
+| HTTP Status Code | Reason Phrase | Description
+| ---------------- | -------- | -----------
+| 200 | OK | no error |
+| 400 | Bad Request | missing/invalid data
+| 401 | Unauthorized | invalid API key
+| 405 | Method Not Allowed | invalid method (GET)
+| 406 | Not Acceptable | unsupported API version
+| 415 | Unsupported Media Type | missing/invalid content type
+| 422 | Unprocessable Entity | missing/invalid request body
+| 500 | Internal Server Error | default error
+
+The error response should include a json-formatted body with a human-readable `"message"` containing further details about the error. The exact error message is up to the implementer, and additional fields can be provided with further information. For example, if the match request specifies an unsupported API version, the server should respond with `Not Acceptable (406)` and a content body such as:
+
+```
+{
+  "message" : "unsupported API version",
+  "supportedVersions" : [ "0.1", "1.0", "1.1" ]
```

Comment:
I was wondering if we should instead return the supported versions in an accept header, but that might not be the right solution.